mithril.d.ts

```diff
@@ -92,8 +92,8 @@ declare module _mithril {
 		onunload?(evt: Event): any;
 	}
 
-	interface MithrilControllerFunction extends MithrilController {
-		(): any;
+	interface MithrilControllerFunction<T extends MithrilController> {
+		(attributes?: any, ... args : any[]): T;
```

Comment:
Nit: Don't make this generic. When I did before, [it caused problems](https://github.com/lhorie/mithril.js/pull/807). Make the return type `MithrilController` itself.

Also, remove the `attributes` property, as it's not necessary.